From 3f3457e70d9e7d5186fede72c8e6044e50cc1eb8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Antti=20M=C3=A4=C3=A4tt=C3=A4?= <antti.maatta@qt.io>
Date: Wed, 1 Oct 2025 13:43:59 +0300
Subject: [PATCH 273/553] Tracepointgen: Fix dash in include regex

Move it to the end so that it doesn't get mistaken for range.

Task-number: QTBUG-140443
Change-Id: I37ed2f80b1dc6632c97638ba7adb8199c222ae63
Reviewed-by: Samuli Piippo <samuli.piippo@qt.io>
(cherry picked from commit fd0987a8ae4388484182f08e3d886177ec61cc14)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/tools/tracepointgen/parser.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools/tracepointgen/parser.cpp b/src/tools/tracepointgen/parser.cpp
index aa51c5c1b95..4c64c1a4973 100644
--- a/src/tools/tracepointgen/parser.cpp
+++ b/src/tools/tracepointgen/parser.cpp
@@ -504,7 +504,7 @@ void Parser::addIncludesRecursive(const QString &filename, QList<QString> &inclu
         data += line + QLatin1Char(QLatin1Char('\n'));
     }
 
-    QRegularExpression includeMacro(QStringLiteral("#include [\"<]([A-Za-z0-9_./-:]*.h)[\">]"));
+    QRegularExpression includeMacro(QStringLiteral("#include [\"<]([A-Za-z0-9_.:/-]*.h)[\">]"));
     QRegularExpressionMatchIterator i = includeMacro.globalMatch(data);
     while (i.hasNext()) {
         QRegularExpressionMatch match = i.next();
@@ -548,7 +548,7 @@ void Parser::parse(QIODevice &input, const QString &name)
 
     QStringList includes;
 
-    QRegularExpression includeMacro(QStringLiteral("#include [\"<]([A-Za-z0-9_./-:]*.h)[\">]"));
+    QRegularExpression includeMacro(QStringLiteral("#include [\"<]([A-Za-z0-9_.:/-]*.h)[\">]"));
     QRegularExpressionMatchIterator i = includeMacro.globalMatch(data);
     while (i.hasNext()) {
         QRegularExpressionMatch match = i.next();
-- 
2.51.2

